execute/tokendata/observer.go

```diff
@@ -60,7 +60,7 @@ func NewConfigBasedCompositeObservers(
 		// e.g. observers[i] := config.CreateTokenDataObserver()
```

Comment:
`NewConfigBasedCompositeObservers` is an  exported function that returns an unexported type. In general it's not a good approach. Either return interface or exported struct.